### PR TITLE
Add nullable annotations to Identity.Extensions

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
@HaoK / @Tratcher I'm very unfamiliar with this code, but the way I went about annotating this was looking at the results from the EF implementation and using that to model the interface. Outside of a couple of weird ones (like normalizing `Id` producing null because we call `ToString()` this seemed straightforward. Once we resolve the shape, I'll remove the NoWarn and update the PublicApi analyzer files.

File: src/Identity/EntityFrameworkCore/src/RoleStore.cs [hunk 1]
```diff
@@ -83,7 +83,7 @@ public class RoleStore<TRole, TContext, TKey, TUserRole, TRoleClaim> :
     /// </summary>
     /// <param name="context">The <see cref="DbContext"/>.</param>
     /// <param name="describer">The <see cref="IdentityErrorDescriber"/>.</param>
-    public RoleStore(TContext context, IdentityErrorDescriber describer = null)
+    public RoleStore(TContext context, IdentityErrorDescriber? describer = null)
```
Comment (HaoK) [inline src/Identity/EntityFrameworkCore/src/RoleStore.cs:86]: sorta random side question, are we going to attempt to switch to !! for things like context here eventually as well?

File: src/Identity/Extensions.Core/src/Rfc6238AuthenticationService.cs [hunk 2]
```diff
@@ -1,6 +1,8 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#nullable enable
```
Comment (HaoK) [inline src/Identity/Extensions.Core/src/Rfc6238AuthenticationService.cs:4]: is this file special to need an explicit on?

**Review by HaoK — APPROVED** [review]

File: src/Identity/EntityFrameworkCore/src/RoleStore.cs (hunk 1 above)
Comment (pranavkm) [reply src/Identity/EntityFrameworkCore/src/RoleStore.cs:86]: I guess. Using `ArgNullException.ThrowIfNull` which is what `!!` would use is apparently ever so slightly faster, so it makes sense that way. That said, for a newly written code, I'm in favor of not having null checks and relying on the nullability annotations to provide guidance.

File: src/Identity/Extensions.Core/src/Rfc6238AuthenticationService.cs (hunk 2 above)
Comment (pranavkm) [reply src/Identity/Extensions.Core/src/Rfc6238AuthenticationService.cs:4]: It's source included by the test project which does not have nullability enabled: https://github.com/dotnet/aspnetcore/blob/c0575788ecadd6e5cfeb4eab635c13b5fd433d37/src/Identity/test/Identity.FunctionalTests/Microsoft.AspNetCore.Identity.FunctionalTests.csproj#L9

**Review by pranavkm — COMMENTED** [review]
Thanks! I"ll fixup the tests soonish

**Pull request merged**